docs/blog/2019-05-17-improvements-to-schema-customization/index.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Improvements to Schema Customization API - Available in Gatsby 2.4.0
+date: 2019-05-17
+author: Mikhail Novikov
+tags:
+  - schema
+  - graphql
+---
+
+Today we are releasing further improvements to the schema customization that [we've released in version 2.2.0](/blog/2019-03-18-releasing-new-schema-customization). You can use them with Gatsby 2.4.0.
+
+It is now possible to indicate to Gatsby, that you want to add a resolver to an explicitly defined fields. Use extensions like `@link` and `@dateformat` to add default arguments or/and resolvers to fields. In addition, when `@dontInfer` is set, Gatsby will no longer run inference for marked type, allowing one to improve performance for large data sets.
+
+## Summary
+
+After about a month of testing schema customization both here and in pre-release we determined a couple of issues. We set up to do this because we wanted to remove uncertainty in people's schemas, when the data changes. However, the original design allowed some uncertainties anyway. In addition, we have made inferrence a more heavy process, trading performance for consistency and didn't really provide a way to opt out of it completely. To summarize:
+
+- Resolvers and arguments of fields like Date and File was determined by inferred data
+- There was no easy way to use arguments/resolvers to override the above
+- Inferrence was run even when `@dontInfer` flag was on
+- There was no way to control inference outside of SDL, eg in Type Builders
+
+Therefore we have some changes to the way we do inferrence. In addition, we are deprecating some of the features introduced is 2.2.0 and will remove them in Gatsby 3.
+
```

Comment:
Could add a heading here, something like "Updates in Gatsby 2.5.0"? Saying something about "these are the problems we identified, here are the updates we've created to fix them"

This would make it really clear where you've stopped talking about 2.2.0 problems and started talking about 2.5.0 updates.

Reply:
Done in `f338bdd` 👍